addons/mail/static/src/models/rtc_session.js

```diff
@@ -13,6 +13,23 @@ registerModel({
         },
     },
     recordMethods: {
+        /**
+         * Plays the video as some browsers may not support or block autoplay.
+         *
+         * @param {Event} ev
+         */
+        async onVideoLoadedMetaData(ev) {
+            try {
+                await ev.target.play();
+            } catch (error) {
+                if (typeof error === 'object' && error.name === 'NotAllowedError') {
```

Comment:
I don't understand this check on `typeof error`, do you know why it's there?

Reply:
I investigated a bit 🕵️‍♂️

The rationale is: anyone can throw anything, not necessarily an error, so we may want to ensure that the caught object is what was expected before doing anything with it, on the model of what you can do in other languages such as Python.

`typeof error === 'object'` is a bad way to go though, since you can still throw `null` which will fulfil the check but cause the program to crash right after trying to access the `name` property.

Ultimately, I find this line a bit confusing and not very helpful, but if we want to keep it anyway it would be better to replace it with  `error instanceof DOMException` in my opinion.

Reply:
Yeah, pretty much. Current code works as long as no one either throw `null` or forged an object with `name` being `NotAllowedError` that has nothing to do with `DOMException/NotAllowedError`.

All this should be technical detail of action `SoundEffect/play`. There should be no exception bubbling in business code of models.